packages/txservice/src/provider.ts

```diff
@@ -216,167 +280,133 @@ export class ChainRpcProvider {
    *
    * @returns A BigNumber representing the estimated gas value.
    */
-  public estimateGas(transaction: providers.TransactionRequest): ResultAsync<BigNumber, TransactionError> {
-    return this.resultWrapper<BigNumber>(false, async () => {
-      const errors: any[] = [];
-      // TODO: #147 If quorum > 1, we should make this call to multiple providers.
-      for (const provider of this._providers) {
-        let result: string;
-        try {
-          // This call will prepare the transaction params for us (hexlify tx, etc).
-          // TODO: #147 Is there any reason prepare should be called for each iteration?
-          const args = provider.prepareRequest("estimateGas", { transaction });
-          result = await provider.send(args[0], args[1]);
-        } catch (error) {
-          const sanitizedError = parseError(error);
-          // If we get a TransactionReverted error, we can assume that the transaction will fail,
-          // and we ought to just throw here.
-          if (sanitizedError.type === TransactionReverted.type) {
-            throw sanitizedError;
-          } else {
-            errors.push(error);
-            continue;
-          }
-        }
+  public async estimateGas(transaction: providers.TransactionRequest): Promise<BigNumber> {
+    const { gasLimitInflation } = this.chainConfig;
 
-        try {
-          return BigNumber.from(result);
-        } catch (error) {
-          throw new GasEstimateInvalid(result, {
-            error: error.message,
-          });
-        }
-      }
-      if (errors.every((e) => e.type === RpcError.type)) {
-        throw new RpcError(RpcError.reasons.FailedToSend, { errors });
+    return this.execute(false, async (provider: SyncProvider) => {
+      // This call will prepare the transaction params for us (hexlify tx, etc).
+      const args = provider.prepareRequest("estimateGas", { transaction });
+      const result = await provider.send(args[0], args[1]);
+      try {
+        return BigNumber.from(result).add(gasLimitInflation ? BigNumber.from(gasLimitInflation) : 0);
+      } catch (error) {
+        throw new GasEstimateInvalid(result, {
+          error: error.message,
+        });
```

Comment:
would this function cause any problems with the error message returned from potentially reverted transactions when you estimate gas?

Reply:
it's not a perfect system, but in reality the error parsing happens twice - once in SyncProvider and once in ChainRpcProvider.execute (the new wrapper). This is bc we want to do that regardless of whether the error comes from the block in ChainRpcProvider (some logic fail) or from ethers/provider itself.

So either way, the error will be parsed the same way it always has. And when it's called a second time, we'll recognize it as an NxtpError and just pass it up as-is